eng/packages/http-client-csharp/generator/Azure.Generator/src/AzureVisitor.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Generator.CSharp.ClientModel;
+using Microsoft.Generator.CSharp.Providers;
+
+namespace Azure.Generator
+{
+    internal class AzureVisitor : ScmLibraryVisitor
+    {
+        protected override TypeProvider? Visit(TypeProvider type)
+        {
+            if (type is ModelProvider || type is EnumProvider || type is ModelFactoryProvider)
+            {
+                type.Namespace = $"{type.Namespace}.Models";
+                type.Type.Namespace = $"{type.Type.Namespace}.Models";
+            }
```

Comment:
I think this is not enough.
Azure's namespace is determined in this way:
we have a RootNamespace which you could read from the configuration `namespace`,
models, enums, and model factory will be in `RootNamespace.Models` and all others go to `RootNamespace`. when the configuration `model-namespace` is true.
if `model-namespace` is false, everything goes to `RootNamespace`.

Reply:
And we need two namespace assignment because when `TypeProvider` produces a `CSharpType`, the csharp type kind of freezes on namespaces therefore we have to change them both?

Reply:
> I think this is not enough. Azure's namespace is determined in this way: we have a RootNamespace which you could read from the configuration `namespace`, models, enums, and model factory will be in `RootNamespace.Models` and all others go to `RootNamespace`. when the configuration `model-namespace` is true. if `model-namespace` is false, everything goes to `RootNamespace`.

Good to know, but `model-namespace` is not part of Configuration yet?